internal/crypto/verify.go

```diff
@@ -0,0 +1,70 @@
+package crypto
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	neofsecdsa "github.com/nspcc-dev/neofs-sdk-go/crypto/ecdsa"
+)
+
+type signedDataFunc = func() []byte
+
+func verifyECDSASignature(pubBin, sig []byte, signedData signedDataFunc, castPub func(*ecdsa.PublicKey) neofscrypto.PublicKey) (*ecdsa.PublicKey, bool) {
+	pub, err := keys.NewPublicKeyFromBytes(pubBin, elliptic.P256())
+	if err != nil {
+		return nil, false
+	}
+	if !castPub((*ecdsa.PublicKey)(pub)).Verify(signedData(), sig) {
+		return nil, false
+	}
+	return (*ecdsa.PublicKey)(pub), true
+}
+
+func verifyECDSASHA512Signature(pubBin, sig []byte, signedDataFn signedDataFunc) (*ecdsa.PublicKey, bool) {
+	return verifyECDSASignature(pubBin, sig, signedDataFn, func(pub *ecdsa.PublicKey) neofscrypto.PublicKey {
+		return (*neofsecdsa.PublicKey)(pub)
+	})
+}
+
+func verifyECDSARFC6979Signature(pubBin, sig []byte, signedDataFn signedDataFunc) (*ecdsa.PublicKey, bool) {
+	return verifyECDSASignature(pubBin, sig, signedDataFn, func(pub *ecdsa.PublicKey) neofscrypto.PublicKey {
+		return (*neofsecdsa.PublicKeyRFC6979)(pub)
+	})
+}
+
+func verifyECDSAWalletConnectSignature(pubBin, sig []byte, signedDataFn signedDataFunc) (*ecdsa.PublicKey, bool) {
+	return verifyECDSASignature(pubBin, sig, signedDataFn, func(pub *ecdsa.PublicKey) neofscrypto.PublicKey {
+		return (*neofsecdsa.PublicKeyWalletConnect)(pub)
+	})
+}
+
+// VerifyTokenSignature verifies t signature.
+func VerifyTokenSignature[T interface {
```

Comment:
https://github.com/nspcc-dev/neofs-node/issues/2795?

Reply:
yes, it relates, I'll note in the commit. For now this API is some kind of experimental, so I'm not ready to place this code in the SDK. And for now it's easier to work within current module only